core/modules/filters/kin.js

```diff
@@ -0,0 +1,107 @@
+/*\
+title: $:/core/modules/filters/kin.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator that recursively finds kindred between tiddlers
+
+\*/
+(function () {
+
+  /*jslint node: true, browser: true */
+  /*global $tw: false */
+  "use strict";
+
+  // TODO: Should I set global tw to true?
+
+  function collectFamilyMembers(tiddler, title, fieldname, direction) {
+    var family_members_from = [],
+      family_members_to = [];
+
+    function addToResultsIfNotFoundAlready(list, title) {
+      if (list.includes(title)) {
+        return false;
+      }
+      list.push(title);
+      return true
+    }
+
+    function findKindredFrom(tiddler, title) {
+      if (addToResultsIfNotFoundAlready(family_members_from, title)) {
+        if (tiddler) {
+          tiddler.getFieldList(fieldname).forEach(function (target_title) {
+            findKindredFrom($tw.wiki.getTiddler(target_title), target_title);
+          });
+        }
+      }
+    }
+
+    function findKindredTo(title) {
+      if (addToResultsIfNotFoundAlready(family_members_to, title)) {
+        $tw.wiki.findListingsOfTiddler(title, fieldname).forEach(function (target_title) {
+          findKindredTo(target_title);
+        });
+      }
+    }
+
+    if ((direction === 'from') || (direction === 'with')) {
+      findKindredFrom(tiddler, title);
+    }
+    if ((direction === 'to') || (direction === 'with')) {
+      findKindredTo(title);
+    }
+    return uniqueArray(family_members_from.concat(family_members_to));
+  }
+
+  // TODO: Is there a better way for unique?
+  function uniqueArray(input) {
+    var seen = {},
+      output = [],
+      len = input.length,
+      j = 0;
+    for (var i = 0; i < len; i++) {
+      var item = input[i];
+      if (seen[item] !== 1) {
+        seen[item] = 1;
+        output[j++] = item;
+      }
+    }
+    return output;
+  }
+
+  /*
+    Export our filter function
+
+    TODO: May I add tests? (editions/test/tiddlers/tests)
+    */
+  exports.kin = function (source, operator, options) {
+    var results = [],
+      needs_exclusion = operator.prefix === '!',
+      suffix_list = (operator.suffix || '').split(':'),
```

Comment:
The PR #3502 adds support for a new `operator.suffixes` structure for standardised parsing of the suffix. Can it be used here?